src/tss2-tcti/tcti-cmd.c

```diff
@@ -442,6 +442,7 @@ TSS2_RC tcti_cmd_get_poll_handles (TSS2_TCTI_CONTEXT *tctiContext,
             LOG_ERROR ("Could not get fileno: %s", strerror (errno));
             return TSS2_TCTI_RC_IO_ERROR;
         }
+        handles->events = POLLIN | POLLOUT | POLLHUP;
```

Comment:
POLLHUP is only returned via revents in the struct pollfd. So you can drop that. POLLOUT might not give you a non-blocking result. If you write larger than available you'll block unless O_NONBLOCK is set in open(2). Definitely need POLLIN.